src/Bitmap/BitmapFile.cpp

```diff
@@ -21,14 +21,14 @@ BitmapFile BitmapFile::CreateIndexed(uint16_t bitCount, uint32_t width, uint32_t
 	return bitmapFile;
 }
 
-BitmapFile BitmapFile::CreateIndexed(uint16_t bitCount, uint32_t width, uint32_t height, const std::vector<Color>& palette)
+BitmapFile BitmapFile::CreateIndexed(uint16_t bitCount, uint32_t width, uint32_t height, std::vector<Color> palette)
 {
 	if (palette.size() > std::size_t(1) << bitCount) {
 		throw std::runtime_error("Unable to create bitmap. Provided palette length is greater than provided bit count.");
 	}
 
 	auto bitmapFile = BitmapFile::CreateIndexed(bitCount, width, height);
-	std::copy(palette.begin(), palette.end(), bitmapFile.palette.begin());
+	std::move(palette.begin(), palette.end(), bitmapFile.palette.begin());
```

Comment:
Ahh, I was thinking:
```cpp
bitmapFile.palette = std::move(palette);
```

I didn't realize there were two versions of `std::move`:
- [`std::move` from `<utility>`](https://en.cppreference.com/w/cpp/utility/move)
- [`std::move` from `<algorithm>`](https://en.cppreference.com/w/cpp/algorithm/move)

Reply:
Yeah, I just read up on the difference a little bit. 

I didn't want to use `bitmapFile.palette = std::move(palette);` as recommended because it opens a discussion about if you can initialize the bitmap with less than a full palette. I think that may be fine, but I wasn't ready to work through some of the implications where this solution will keep the palette at full length of the bit field even if a smaller palette is passed in.

I think I will merge as is. @DanRStevens, no problem if you want to work an improved implementation or writeup an improvement in an issue, but I don't want to stray too far from the tileset conversion work right now.

Reply:
That's fine. We may want to revisit later though. Especially in regards to potentially uninitialized palette entries in case a smaller palette is passed in. Or possibly referencing non-existent palette entries if a shorter palette is stored. We may want to refer back to the bitmap file format to see what is supported there.